dvc/tree/dropbox.py

```diff
@@ -0,0 +1,289 @@
+import datetime
+import json
+import logging
+import os
+import threading
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+
+from funcy import cached_property, wrap_prop
+
+from dvc.hash_info import HashInfo
+from dvc.path_info import CloudURLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+ACCESS_TOKEN = "DROPBOX_ACCESS_TOKEN"
+EXPIRATION = "DROPBOX_EXPIRES_AT"
+REFRESH_TOKEN = "DROPBOX_REFRESH_TOKEN"
+
+
+Auth = namedtuple("Auth", ["client", "creds"])
+
+
+class DropboxCredProvider(metaclass=ABCMeta):
+    DROPBOX_APP_KEY = ""
+    DROPBOX_APP_SECRET = ""
+
+    @abstractmethod
+    def can_auth(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def _gather_auth(self):
+        raise NotImplementedError
+
+    def auth(self):
+        import dropbox
```

Comment:
@efiop do we have a mechanism these days to handle gracefully the failure? (and recommend installing it with a specific dependency)?

Reply:
Sure, this needs REQUIRES https://github.com/iterative/dvc/blob/dca48fe8d7f86e49dad856bad1e970b837912793/dvc/tree/s3.py#L24 and the rest will be handled automatically.

Reply:
It already is addressed [here](https://github.com/iterative/dvc/pull/4793/files#diff-d0bb1fd2cb5f725432045fae5b6d0fc2cffe67b24cff73cc56210379e356401fR147).